### ISSUE TITLE
[YSQL] Improve error message when split clause fails on hash-partitioned index

### ISSUE THREAD

**Issue by kripasreenivasan** [yugabyte/yugabyte-db#16213]
Jira Link: [DB-5642](https://yugabyte.atlassian.net/browse/DB-5642)
### Description

The error when specifying SPLIT AT VALUES clause on a hash-partitioned index can be improved
```
cl_db_001=# create index ind10 on table_1(id) SPLIT AT VALUES ((100));
ERROR:  SPLIT AT option is not yet supported for hash partitioned tables
```

```
cl_db_001=# \d+ table_1
                                                              Table "public.table_1"
    Column     |            Type             | Collation  | Nullable |               Default               | Storage  | Stats target | Description
---------------+-----------------------------+------------+----------+-------------------------------------+----------+--------------+-------------
 id            | text                        |            | not null |                                     | extended |              |
 uuid_col      | uuid                        |            |          | uuid_generate_v4()                  | plain    |              |
 name          | text                        | en_US.utf8 |          |                                     | extended |              |
 c             | complex                     |            |          |                                     | extended |              |
 info          | json                        |            |          |                                     | extended |              |
 contact       | jsonb                       |            |          |                                     | extended |              |
 arr           | smallint[]                  |            |          |                                     | extended |              |
 cash          | money                       |            |          |                                     | plain    |              |
 i             | inet                        |            |          |                                     | main     |              |
 m             | macaddr                     |            |          |                                     | plain    |              |
 i2            | integer                     |            | not null | nextval('table_1_i2_seq'::regclass) | plain    |              |
 i3            | bigint                      |            | not null | nextval('table_1_i3_seq'::regclass) | plain    |              |
 val           | smallint                    |            |          |                                     | plain    |              |
 details       | e_details                   |            |          |                                     | plain    |              |
 age           | integer                     |            |          |                                     | plain    |              |
 collated_data | text                        | POSIX      |          |                                     | extended |              |
 date          | date                        |            |          |                                     | plain    |              |
 n             | numeric(3,2)                |            |          |                                     | main     |              |
 r             | real                        |            |          |                                     | plain    |              |
 c1            | character(1)                |            |          |                                     | extended |              |
 created_at    | timestamp with time zone    |            |          |                                     | plain    |              |
 uuid0         | uuid                        |            |          | uuid_nil()                          | plain    |              |
 uuid1         | uuid                        |            |          | uuid_generate_v1()                  | plain    |              |
 p1            | point                       |            |          |                                     | plain    |              |
 t1            | time without time zone      |            |          |                                     | plain    |              |
 ts1           | timestamp without time zone |            |          |                                     | plain    |              |
 i4            | interval                    |            |          |                                     | plain    |              |
 p2            | path                        |            |          |                                     | extended |              |
 p3            | polygon                     |            |          |                                     | extended |              |
 b             | box                         |            |          |                                     | plain    |              |
 c2            | circle                      |            |          |                                     | plain    |              |
 l             | line                        |            |          |                                     | plain    |              |
 l1            | lseg                        |            |          |                                     | plain    |              |
 a2            | text[]                      |            |          |                                     | extended |              |
 zip           | postal_code                 |            |          |                                     | extended |              |
Indexes:
    "table_1_pkey" PRIMARY KEY, lsm (id HASH)
Options: colocation=false
```

[DB-5642]: https://yugabyte.atlassian.net/browse/DB-5642?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

**Comment by yifanguan** [comment]
"table_1_pkey" PRIMARY KEY, lsm (id HASH)
`table_1` is hash-partitioned. However, this is still an issue because a secondary index has its own sharing strategy.
An index can use range sharding even if its base table uses hash sharding strategy.

**Comment by yifanguan** [comment]
"table_1_pkey" PRIMARY KEY, lsm (id HASH)
table_1 is hash-partitioned.

**Comment by yifanguan** [comment]
One improvement we can do is to let the error message more friendly.
In this case, output `ERROR:  SPLIT AT option is not yet supported for hash partitioned indexes` instead of `ERROR:  SPLIT AT option is not yet supported for hash partitioned tables`

**Comment by sidjha1** [comment]
Issue resolved by commit `1bf0dab`
